src/DotNetty.Transport/Channels/ChannelOutboundBuffer.cs

```diff
@@ -379,11 +387,25 @@ public List<ArraySegment<byte>> GetSharedBufferList()
                                 // of Object allocation
                                 entry.Buffers = nioBufs = buf.GetIoBuffers();
                             }
-                            foreach (ArraySegment<byte> b in nioBufs)
+                            for (int i = 0; i < nioBufs.Length && nioBufferCount < maxCount; ++i)
```

Comment:
`i++` please. I suppose it won't change the meaning but does confuse for a second the way it is right now